versions/2.0.0

jeswr opened the pull request:
WIP For version 2 of N3.js

TODO:
 - Make lists and similar entites work in arbitrary path positions (in N3)
 - Predicate literals?
 - Error on subject lists in turtle?

jeswr committed:
chore: add nested list test

jeswr committed:
fix: support quoted triples in list

jeswr committed:
breaking: drop support for quads in quoted triples as they are forbidden in the rdf-star spec

jeswr committed:
feat: support annotated triples

jeswr committed:
chore: error on quoted compound bnodes

jeswr committed:
feat: turtle-star spec tests are passing

jeswr committed:
chore: fix lint and coverage errors

jeswr committed:
chore: remove commented code

jeswr committed:
chore: rename RDF* -> RDF-star

jeswr committed:
chore: update RDF-star reference in readme

jeswr committed:
chore: fix round trip on deeply nested rdfstar triples

jeswr committed:
chore: add tests from https://github.com/rdfjs/N3.js/pull/303

jeswr committed:
chore: describe quoted triple predicate parsing

Co-authored-by: Ruben Verborgh <ruben@verborgh.org>

jeswr committed:
chore: clarify use of graph term in quoted quads

jeswr committed:
fix: allow a split between '|' and '}' (see https://github.com/rdfjs/N3.js/pull/311#discussion_r1060687020)

jeswr committed:
chore: remove doubling comment

jeswr committed:
chore: add comment about nested parameter

jeswr committed:
fix: use describe for all shouldParse test suites

jeswr committed:
fix: dont interpret }| as {|

jeswr committed:
perf: mint quad ids using term ids

jeswr committed:
Update test/N3Parser-test.js

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

jeswr committed:
Update src/N3Parser.js

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

jeswr committed:
Update src/N3Parser.js

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

jeswr committed:
fix: fix broken N3Store tests

jeswr committed:
chore: improve tests range

jeswr committed:
chore: use _termToNumericId to convert Ids

jeswr committed:
chore: don't mint new ids unecessarily

jeswr committed:
feat: allow nested graph terms

jeswr committed:
Update src/N3Parser.js

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

jeswr committed:
Update src/N3Parser.js

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

jeswr committed:
chore: add performance testing

jeswr committed:
chore: add performance test for limited annotations

jeswr committed:
chore: document writing rdf-star

jeswr committed:
BREAKING CHANGE: enable rdfStar support by default

jeswr committed:
chore: update docs to not rdfStar default

jeswr committed:
chore: refactor lexer

jeswr committed:
fix: re-enable line mode check

jeswr committed:
chore: refactor lexer

jeswr committed:
fix: make rdf-star work with n3 paths

jeswr committed:
Merge branch 'perf/better-quad-ids' into feat/rdfstar-support

jeswr commented:
Note to self for when I revisit in the morning - the matrix based rdf-star tests appear to be passing even on cases they should be failing and erroring.

I need to investigate.

jeswr committed:
fix: fix support for paths to support all usages supported by the N3 spec

TallTed commented on README.md:
```diff
@@ -358,16 +370,14 @@ The N3.js parser and writer is fully compatible with the following W3C specifica
-If a format string contains `star` or `*`
-(e.g., `turtlestar` or `TriG*`),
-RDF* support for that format will be enabled.
+To disable RDF-star support pass `rdfStar` to `false` in the constructor.
```
```suggestion
To disable RDF-star support, pass `rdfStar` to `false` in the constructor.
```

jeswr committed:
fix: correctly handle named graphs in lists

* chore: add test cases for named nodes in lists

* chore: add more graph in list test cases

* fix: correctly handle named graphs in lists

jeswr committed:
Update README.md

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

jeswr committed:
fix: support predicates in the list position in Notation3 (#345)

* fix: support predicates in the list position in Notation3

* chore: remove TODOs

jeswr committed:
feat: parse subject and predicate literals in N3 mode (https://github.com/rdfjs/N3.js/pull/337)

jeswr committed:
BREAKING CHANGE: Make class Term constructors internal (#338)

These constructors take the `id` at face value so can be interpreted as the
wrong term type if an invalid value is given; for instance a `NamedNode` can
be interpred as a `Literal` if instantiated with `new NamedNode('"hellow world"')`.

Consumers of the library should use the `DataFactory` instead to create new terms.

jeswr committed:
fix: use BlankNode according to RDF/JS spec (#346)

Co-authored-by: Tomasz Pluskiewicz <tpluscode@users.noreply.github.com>

jeswr committed:
BREAKING CHANGE: parse <= as log:isImpliedBy (https://github.com/rdfjs/N3.js/pull/327)

jeswr commented:
Before releasing we should make sure that our mime types for rdf-star are correct.

jeswr commented:
Closing this out: v2.0.0 shipped in January 2026 via a different route — aa6a469 replaced RDF-star support with RDF 1.2 (triple terms, reified triples, directional language tags), so the RDF-star groundwork this branch was built on (quoted triples in lists, annotation syntax, star spec tests, the rdfStar default) is now moot. The remaining pieces landed on main independently: annotated triples (#358), numeric term ids in the store (#369), the BlankNode RDF/JS fix, and `<=` as log:isImpliedBy behind a feature flag (#500).

A few Notation3-mode improvements only ever landed on this branch and are still worth having — each now has a tracker against current main: predicates in list position (#345 → #619), subject/predicate literals in N3 mode (#337 → #328, fix up in #613), named graphs in lists (#342, fix up in #617), and the N3 path fixes (#348). Those are being extracted as small, focused PRs rather than rebasing this branch. The Term-constructor visibility change (#338) is semver-major and is recorded with #313 for a future major.

The versions/2.0.0 branch stays on the org for reference.
